src/app/login/login.component.html

```diff
@@ -3,20 +3,36 @@
         <b><mat-card-subtitle>Login</mat-card-subtitle></b>
         <mat-divider></mat-divider>
         <mat-card-content>
+          <form [formGroup]="loginFormGroup" #ngForm>
                 <div class="input-class">
                   <mat-form-field class="full-width">
-                    <input matInput placeholder="Enter Your Team Name" [(ngModel)]="teamName"><mat-icon class="prefix-icon">person</mat-icon>
+                    <input [formControl]="teamnameFormControl" matInput placeholder="Enter Your Team Name" [(ngModel)]="teamName" name="name" required><mat-icon class="prefix-icon">person</mat-icon>
+                    <mat-error *ngIf="teamnameFormControl.hasError('required')">teamname is required.</mat-error>
```

Comment:
Why you don't use `detailsValid` function in `login.component.ts` and put the return value instead 2 if statements?